crypto/rand/drbg_lib.c

```diff
@@ -281,6 +280,10 @@ int rand_drbg_restart(RAND_DRBG *drbg,
 
             /* will be picked up by the rand_drbg_get_entropy() callback */
             drbg->pool = RAND_POOL_new(entropy, len, len);
+
+            if (drbg->pool == NULL)
+                return 0;
```

Comment:
We should add a RANDerr(RAND_F_RAND_DRBG_RESTART, ERR_R_MALLOC_FAILURE) before the return.

Reply:
and delete the blank line at 283.

Reply:
@kaduk There is more than one location where a failure of `RAND_POOL_new()` is not reported. This was done on purpose, because `RAND_POOL_new()` already reports the error and I did not want to duplicate the message. 
```C
RAND_POOL *RAND_POOL_new(int entropy, size_t min_len, size_t max_len)
{
    RAND_POOL *pool = OPENSSL_zalloc(sizeof(*pool));

    if (pool == NULL) {
        RANDerr(RAND_F_RAND_POOL_NEW, ERR_R_MALLOC_FAILURE);
        goto err;
    }
```
Please let me now if a single error message in `RAND_POOL_new()`is sufficient or if you still prefer to have an error message in the calling function, too. Then I will correct all locations.

Reply:
@richsalz: This is not the only location where I have this extra line after the `RAND_POOL_new()` call. I will correct all of them.

Reply:
Ah, sorry, I was just looking at the last commit -- if RAND_POOL_new() is adding to the error stack then we're fine.